### PR TITLE
Fix `prefer-object-spread` in `src/config`

### PR DESCRIPTION
**Pull request by fisker**
Part of #7339

### CODE REVIEW

File: src/config/resolve-config.js [hunk 1]
```diff
@@ -126,13 +125,13 @@ resolveConfigFile.sync = filePath => {
 };
 
 function mergeOverrides(configResult, filePath) {
-  const options = Object.assign({}, configResult.config);
-  if (filePath && options.overrides) {
+  const { overrides, ...options } = { ...configResult.config };
```
Comment (fisker) [inline src/config/resolve-config.js]: Notice: little refactor here.

File: src/config/resolve-config.js [hunk 2]
```diff
@@ -74,11 +74,10 @@ function _resolveConfig(filePath, opts, sync) {
...
 
   const unwrapAndMerge = ([result, editorConfigured]) => {
-    const merged = Object.assign(
-      {},
-      editorConfigured,
-      mergeOverrides(Object.assign({}, result), filePath)
-    );
+    const merged = {
+      ...editorConfigured,
+      ...mergeOverrides({ ...result }, filePath)
```
Comment (thorn0) [inline src/config/resolve-config.js]:
```suggestion
      ...mergeOverrides(result, filePath)
```
`mergeOverrides` doesn't modify this argument.

Comment (fisker) [reply src/config/resolve-config.js]: Done.

File: src/config/resolve-config.js [hunk 3]
```diff
@@ -126,13 +125,11 @@ resolveConfigFile.sync = filePath => {
...
 function mergeOverrides(configResult, filePath) {
-  const options = Object.assign({}, configResult.config);
-  if (filePath && options.overrides) {
-    const relativeFilePath = path.relative(
-      path.dirname(configResult.filepath),
-      filePath
-    );
-    for (const override of options.overrides) {
+  const { config } = configResult || {};
+  const { overrides, filepath: configPath, ...options } = config || {};
```
Comment (thorn0) [inline src/config/resolve-config.js]: It was `configResult.filepath`, not `configResult.config.filepath`.

Comment (fisker) [reply src/config/resolve-config.js]: Fixed, thanks.

**Pull request merged**